fastapi/security/utils.py

```diff
@@ -8,3 +14,25 @@ def get_authorization_scheme_param(
         return "", ""
     scheme, _, param = authorization_header_value.partition(" ")
     return scheme, param
+
+
+_SecurityDepFunc = TypeVar(
+    "_SecurityDepFunc", bound=Callable[[Any, HTTPConnection], Awaitable[Any]]
+)
+
+
+def handle_exc_for_ws(func: _SecurityDepFunc) -> _SecurityDepFunc:
+    @wraps(func)
+    async def wrapper(self: Any, request: HTTPConnection) -> Any:
+        try:
+            return await func(self, request)
+        except HTTPException as e:
+            if not isinstance(request, WebSocket):
+                raise e
+            # close before accepted with result a HTTP 403 so the exception argument is ignored
+            # ref: https://asgi.readthedocs.io/en/latest/specs/www.html#close-send-event
+            raise WebSocketException(
+                code=WS_1008_POLICY_VIOLATION, reason=e.detail
+            ) from None
+
+    return wrapper  # type: ignore
```

Comment:
With the new Starlette version (0.37.0), you don't need this.

Reply:
Thanks for the confirmation!
So, we need to wait until FastAPI is updated to use Startlette 0.37. Then we will need to exclude all the stuff with `@handle_exc_for_ws` from this PR, change tests and merge it.

Reply:
Sure, we'll wait for PR #11146 to be completed. 

By the way, could you provide details on the change in Starlette 0.37 that enables raising HTTPException in WebSocket sessions?